UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareServiceCache.cs

```diff
@@ -125,18 +140,23 @@ private Task<BatchDetail> CheckIfCacheProductsExistsInBlob(string exchangeSetRoo
                         if (!fileSystemHelper.CheckFileExists(path))
                         {
                             CloudBlockBlob cloudBlockBlob = await azureBlobStorageClient.GetCloudBlockBlob(fileName, storageConnectionString, internalBatchDetail.BatchId);
-                            
+
                             //Added to check blob exception
                             try
                             {
                                 await fileSystemHelper.DownloadToFileAsync(cloudBlockBlob, path);
+                                updateNumbers.Add(itemUpdateNumber.Value);
+                            }
+                            catch (StorageException storageEx) when (storageEx.Message.Contains("The specified blob does not exist"))
```

Comment:
Meant to raise this last time, but in the StorageException block, there is no rethrow of the exception like there is with the more general Exception block. So the code will log an error, then continue processing - can I confirm this is correct?

Reply:
Yes, We will just log it and continue processing.